cli/pkg/github/auth.go

```diff
@@ -0,0 +1,140 @@
+package github
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+// DeviceFlow allows to authorize users to GitHub.
+// See https://docs.github.com/en/developers/apps/authorizing-oauth-apps#device-flow.
+type DeviceFlow struct {
+	baseURI *url.URL
+	client  *http.Client
+}
+
+// NewDeviceFlow creates a new GitHub device authorization flow.
+func NewDeviceFlow(baseURL *url.URL, client *http.Client) DeviceFlow {
+	return DeviceFlow{baseURI: baseURL, client: client}
+}
+
+// GetCodes requests the device and user verification codes from GitHub.
+func (df DeviceFlow) GetCodes(req *GetCodesRequest) (*GetCodesResponse, error) {
+	uri := df.uri("login/device/code")
```

Comment:
Would you mind adding this path to `cli/pkg/github/github.go` and use it as a constant as well? By doing this, if it changes, we'd need to simply change the value in `github.go`. Same as the `baseURL` variable.